main.c

```diff
@@ -205,27 +230,86 @@ static int dbus_event(int fd, uint32_t mask, void *data) {
 	return count;
 }
 
-static void setup_sleep_listener(void) {
+static void connect_to_bus(void) {
 	int ret = sd_bus_default_system(&bus);
 	if (ret < 0) {
 		errno = -ret;
 		swayidle_log_errno(LOG_ERROR, "Failed to open D-Bus connection");
 		return;
 	}
+	struct wl_event_source *source = wl_event_loop_add_fd(state.event_loop,
+		sd_bus_get_fd(bus), WL_EVENT_READABLE, dbus_event, bus);
+	wl_event_source_check(source);
+}
 
-	ret = sd_bus_match_signal(bus, NULL, "org.freedesktop.login1",
+static void setup_sleep_listener(void) {
+	int ret = sd_bus_match_signal(bus, NULL, "org.freedesktop.login1",
                 "/org/freedesktop/login1", "org.freedesktop.login1.Manager",
                 "PrepareForSleep", prepare_for_sleep, NULL);
 	if (ret < 0) {
 		errno = -ret;
-		swayidle_log_errno(LOG_ERROR, "Failed to add D-Bus signal match");
+		swayidle_log_errno(LOG_ERROR, "Failed to add D-Bus signal match : sleep");
 		return;
 	}
 	acquire_sleep_lock();
+}
 
-	struct wl_event_source *source = wl_event_loop_add_fd(state.event_loop,
-		sd_bus_get_fd(bus), WL_EVENT_READABLE, dbus_event, bus);
-	wl_event_source_check(source);
+static void setup_lock_listener(void) {
```

Comment:
There's some overlap between these functions, could probably share some code

Reply:
you mean setup_lock_listener, setup_unlock_listener ?

yeah, a bit... but I would be suprised if there were a third user, which would make sharing worth it...
i'll see what I can do anyway, but It won't share more than a couple of lines

Reply:
Could share the `GetSessionByPID` stuff